### PR TITLE
Update docs to reflect newest releases

### PR DESCRIPTION
**Pull request by jpreese**
This goes through the current documentation and updates them to reflect the latest changes to Conftest (rendered output, flags, etc).

Documentation of note would be the new `--combine` behavior and `--fail-on-warn`.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,13 +24,13 @@ Before submitting large changes, please open an issue on GitHub outlining:
 - Detailed description of what your changes would entail.
 - Alternative solutions or approaches if applicable.
 
-Use your judgement about what constitutes a large change. If you aren't sure, send a message to the `#conftest` channel in the OPA slack or submit an issue on GitHub.
+Use your judgment about what constitutes a large change. If you aren't sure, send a message to the `#conftest` channel in the OPA slack or submit an issue on GitHub.
```
Comment (Blokje5) [inline CONTRIBUTING.md:27]: Typo here, should be judgement right?

File: docs/index.md [hunk 2]
```diff
@@ -1,65 +1,68 @@
...
-Conftest is a utility to help you write tests against structured configuration data. For instance you could write tests for your Kubernetes configurations, or Tekton pipeline definitions, Terraform code, Serverless configs or any other structured data.
+Conftest is a utility to help you write tests against structured configuration data. For instance, you could write tests for your Kubernetes configurations, Tekton pipeline definitions, Terraform code, Serverless configs or any other structured data.
 
 Conftest relies on the Rego language from [Open Policy Agent](https://www.openpolicyagent.org/) for writing the assertions. You can read more about Rego in [How do I write policies](https://www.openpolicyagent.org/docs/how-do-i-write-policies.html) in the Open Policy Agent documentation.
 
 ## Usage
 
-Conftest allows you to write policies using Open Policy Agent/rego and apply them to one or
-more configuration files. Policies by default should be placed in a directory called `policy` but this can be overridden.
+Policies by default should be placed in a directory called `policy`, but this can be overridden with the `--policy` flag.
```
Comment (Blokje5) [inline docs/index.md:9]: Maybe it would be good to introduce the concept of policies a bit more, as above they are referred to as assertions.

**Review by Blokje5 — CHANGES_REQUESTED** [review]
Two small comments, but besides that nice to see some consistency improvements in the documentation!

File: CONTRIBUTING.md (hunk 1 above)
Comment (jpreese) [reply CONTRIBUTING.md:27]: LOL I actually Google'd this last night. Judgment _appears_ to be the "more correct" approach according to what I'm seeing, but in the end both are acceptable and didn't necessarily warrant a change. I just didn't bother going back and undoing it.

File: docs/index.md (hunk 2 above)
Comment (jpreese) [reply docs/index.md:9]: Good catch, I do prefer the consistency to keep calling them policies. I also updated the link to the documentation site.

**Review by Blokje5 — APPROVED** [review]
LGTM!

**Pull request merged**